ipapython/ipaldap.py

```diff
@@ -735,6 +787,46 @@ def __init__(self, ldap_uri, start_tls=False, force_schema_updates=False,
 
         self._conn = self._connect()
 
+        with self.error_handler():
+            if demand_cert:
+                ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, True)
+                self.conn.set_option(ldap.OPT_X_TLS_DEMAND, True)
+
+            if sasl_nocanon:
+                self.conn.set_option(ldap.OPT_X_SASL_NOCANON, ldap.OPT_ON)
+
+            if start_tls:
+                self.conn.start_tls_s()
```

Comment:
This block should either be moved to `_connect()`, or the rest of `_connect()` should be moved here.